docs/learn/tour/tables.md

```diff
@@ -1,209 +1,307 @@
-q)ec1
-city         country pop
------------------------------
-Istanbul     Turkey  15067724
-Moscow       Russia  12615279
-London       UK      9126366
-StPetersburg Russia  5383890
-```
+In-memory tables live RAM of a process. This memory is volatile and limited in size but allows for fast data retrieval. Typically you will be in a position to generate and access a maximum of 10s of gigabytes of data. As the data stored in RAM is volatile until the data is persisted any exiting of a process will result in the data being lost and will require the data to be regenerated, a task which may be complex.
 
-!!! tip "Equals means equals"
+### What is a Table?
 
-    In q names are assigned values with the colon.
-    The equals sign `=` is the Equals operator. It returns a boolean.
+At its most basic a table is a mapping between a list of column names which each have an associated list of corresponding column values. This column to list mapping form the basis behind why q tables are described as being column-oriented, this is in contrast to row-oriented tables in most relational databases. Additionally because lists are ordered so are the content of a column.
 
-    ```q
-    q)a:5
-    q)a+2   / a gets 5
-    7
-    q)a=2   / no, it is not
-    0b
-    ```
+These two points (column oriented and ordered) in particular help to make kdb+/q extremely efficient at storing, manipulating and retrieving sequential data, which is critically important to interactions with time-series data.
 
-Unlike classical relational databases, q tables are ordered. You can index them.
-A table is a list of dictionaries. Any single row is a dictionary.
+We can generate tables in two ways which illustrate this direct mapping:
 
-```q
-q)ec1 2
-city   | `London
-country| `UK
-pop    | 9126366
-```
+1. Creating a table from a dictionary
 
-<!-- :fontawesome-regular-hand-point-right:
-[Dictionaries](dictionaries.md)
- -->
-And a list of dictionaries with the same keys is – a table.
+	```q
+	q)dict:`items`sales`prices!(`nut`bolt`cam`cog;6 8 0 3;10 20 15 20)
+	items | nut bolt cam cog
+	sales | 6   8    0   3  
+	prices| 10  20   15  20
+	q)tab:flip dict
+	q)tab
+	items sales prices
+	------------------
+	nut   6     10
+	bolt  8     20
+	cam   0     15
+	cog   3     20
+	```
 
-```q
-q)ec1 2 0
-city     country pop
--------------------------
-London   UK      9126366
-Istanbul Turkey  15067724
-```
+2. Creating a table by specifying column names and initial values explicitly
 
-Flipping a table gets you its columns as a dictionary of vectors.
+	```q
+	q)tab2:([]items:`nut`bolt`cam`cog;sales:6 8 0 3;prices:10 20 15 20)
+	q)tab~tab2
+	1b
+	```
 
-```q
-q)flip ec1
-city   | Istanbul Moscow   London  StPetersburg
-country| Turkey   Russia   UK      Russia
-pop    | 15067724 12615279 9126366 5383890
-```
+The form for the second method, for a table with j primary keys and n columns in total, is:
+
+<div markdown="1" class="typewriter">
+t:([] c~1~:v~1~; … ;c~n~:v~n~)
+</div>
```

Comment:
something is not right here

Reply:
Once rendered this provides the mathematical definition so should be fine

![Screenshot 2024-12-13 at 09 43 04](https://github.com/user-attachments/assets/dcac4c46-7af6-4310-9ad6-98b0197e5ebf)

Reply:
aaah, ok ok. Preview mode is not very helpful. Apologies, I'm an idiot. :(